data_utils/ds_addmult_mod10_helpers.py

```diff
@@ -99,16 +139,25 @@ def build_dataset_addmult_mod10(
         'test': test,
     })
 
-    tokenizer = CharVocabulary(chars=set('0123456789+*()='))
-    remove_columns = ['height', 'width', 'example', 'answer', 'ans_mod10']
-    if lm_mode:
+    tokenizer = CharVocabulary(chars=set('0123456789+*()[]xyzL=_'), ignore_char='_', ignore_char_idx=0)
+    remove_columns = ['height', 'width', 'example', 'answer', 'ans_mod10', 'ans_sublabels', 'tree_sig']
+    if lm_mode and not use_intermediates:
         dataset = dataset.map(lambda example, idx: {  
             'in': tokenizer(example['example'] + '=' + str(example['ans_mod10'])),
             'in_len': len(tokenizer(example['example'] + '=' + str(example['ans_mod10']))),
             'idxs': idx,
             'string': example['example']
         }, with_indices=True,
         remove_columns=remove_columns)
+    elif lm_mode and use_intermediates:
+        dataset = dataset.map(lambda example, idx: {  
+            'in': tokenizer(example['example']),
+            'in_len': len(tokenizer(example['example'])),
+            'idxs': idx,
+            'labels': tokenizer(str(example['ans_sublabels'])),
+            'labels_len': len(str(example['ans_sublabels'])) - str(example['ans_sublabels']).count('_'), # number of close brackets
```

Comment:
Nit: Best practice to not hardcode `'_'` as ignore token and instead use the ignore char from line 142, just in case we want to change this character in the future.

line 142:
```
 ignore_char = '_'
 tokenizer = CharVocabulary(chars=set('0123456789+*()[]xyzL=_'), ignore_char=ignore_char, ignore_char_idx=0)
```
```suggestion
            'labels_len': len(str(example['ans_sublabels'])) - str(example['ans_sublabels']).count(ignore_char), # number of close brackets
```

Reply:
Good catch ty!